Added Google OAuth for login and registering purposes!

LitZeus opened the pull request:
### Implemented the Google OAuth authentication in the login and sign up pages

### following are the snapshots of it!

![376067593-08aca253-4e6f-4b4c-a300-cf31181bf941](https://github.com/user-attachments/assets/3065d61d-3a20-47e6-a512-d72d43b1ec79)
![376067749-40f54293-1813-49ce-ae0e-de31f9803088](https://github.com/user-attachments/assets/4c49eed4-15ce-488e-b495-1dac203c1f6e)
![376067606-d7c1de39-d968-4072-9641-7ddcc5528c20](https://github.com/user-attachments/assets/e854fbaa-fde2-4eed-a0a7-81b5eccd84d4)

SaranshBangar reviewed (changes requested):
Please resolve the deployment error

![image](https://github.com/user-attachments/assets/514ac234-d2b9-475d-b33b-d981b104d930)

LitZeus commented:
@SaranshBangar Have you checked the environment variables (.env file) as I have updated it and added firebase application variables. You can check the application name in the snapshot given below which I have created for testing using my own firebase credentials.
I have also updated the `README.md` according to the new variables.
also I will remove the quotes and the blank spaces as it could also be the reason for the error.
Please reflect me after checking it!
<br />

![image](https://github.com/user-attachments/assets/21f493cf-8a5f-4bd0-991a-b7cd1c0d086f)

LitZeus commented:
@SaranshBangar 

Resolved the merge commits! 

Also can i update the .gitignore as the package-lock.json file is also uploading which is not required for the project repo on github and auto generated after the package installing through node package manager.
